querysrv.go

```diff
@@ -329,10 +329,12 @@ func (s *querysrv) handleAnnounce(ctx context.Context, remote net.IP, deviceID p
 		}
 
 		ip := net.ParseIP(host)
-		if len(ip) == 0 || ip.IsUnspecified() {
-			uri.Host = net.JoinHostPort(remote.String(), port)
+		if host == "" || ip.IsUnspecified() {
```

Comment:
Now I think about it, `tcp6://:port` and `tcp6://[::]:port` addresses might not be mapped correctly to IPv6 client address with this implementation..